File: vm/tests/all/interop/simpleClass/Point.cpp
//Point.cpp
#include<polyglot.h>
#include<math.h>

class Point {
	private:
	int x;
	int y;
	public:
	int getX();
	int getY();
	void setX(int dX);
	void setY(int dY);
	double distanceTo(Point* other);
	int checkMe(int dX, int dY);
};

POLYGLOT_DECLARE_CLASS(Point);

int Point::getX() {
	return x;
}

int Point::getY() {
	return y;
}

void Point::setX(int dX) {
	x=dX;
}

void Point::setY(int dY) {
	y=dY;
}

int Point::checkMe(int dX, int dY) {
	return dX==x && dY==y;
}

double distanceTo(Point* other) {
	double dX = x - other->x;
	double dY = y - other->y;
	return sqrt(dX*dX+dY*dY);
}

File: vm/tests/all/interopComplex/simpleClass/Point.py
#Point.py
import math

class Point:
	def __init__(self, x, y):
		self.x=x
		self.y=y
	
	def getX(self):
		return self.x
	
	def getY(self):
		return self.y
	
	def setX(self, val):
		self.x=val

	def setY(self, val):
		self.y=val

	def checkMe(self, x, y):
		return self.x==x and self.y==y
	
	def distanceTo(self, other) {
		dX = self.x - other.x
		dY = self.y - other.y
		return math.sqrt(dX*dX+dY*dY)
	}


